pkg/ruler/manager.go

```diff
@@ -20,6 +20,24 @@ import (
 	"github.com/cortexproject/cortex/pkg/util"
 )
 
+var (
+	configUpdatesTotal = promauto.NewCounterVec(prometheus.CounterOpts{
+		Namespace: "cortex",
+		Name:      "ruler_config_updates_total",
+		Help:      "Total number of config updates triggered by a user",
+	}, []string{"user"})
+	configUpdateFailuresTotal = promauto.NewCounterVec(prometheus.CounterOpts{
+		Namespace: "cortex",
+		Name:      "ruler_config_update_failures_total",
+		Help:      "Total number of config update failures triggered by a user",
+	}, []string{"user", "reason"})
+	userManagerFailed = promauto.NewGaugeVec(prometheus.GaugeOpts{
```

Comment:
I had in mind that we could use the prometheus-style `last_reload_successful` type of thing.

https://github.com/prometheus/prometheus/blob/9438bf735a1e1e968f9403c2a1f606fde9a068a9/cmd/prometheus/main.go#L73-L80

I get the feeling that users might be a bit more familiar with this.

Reply:
[not a blocker to me] I would also be up for a metric with the timestamp. I believe you can alert similarly to the boolean, but it's also more expressive cause you can easily compute since how long it's broken.